### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by shafiuyushawu**
Hello Reviewer

My name is @shafiuyushawu  and this milestone fulfills the following requirements.

### Created the following models
- Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum--rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).
- Set up associations between models.
  - Implemented `author_id` column in posts table should be the foreign_key for the users table.

### Created custom methods
Additional methods created;
- Users
  - A method that returns the 3 most recent posts for a given user.
- Posts
  - A method that updates the posts counter for a user.
  - A method which returns the 5 most recent comments for a given post.
- Comments
  - A method that updates the comments counter for a post.
- Likes
  - A method that updates the likes counter for a post.
- Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  after_save :update_posts_count
+
+  def recent_five_comments
+    comments.order(create_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_posts_count
+    author.increment(:post_counter)
```
Comment (mckent05) [inline app/models/post.rb]: - You have actually done well so far, but there is just a minor issue to be fixed so your project does not have bugs in the future, the increment method is missing the `!`. Please fix this 🙏. It should be ```author.increment!(:post_counter)```

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @shafiuyushawu, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job creating and setting up associations between models 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (shafiuyushawu) [reply app/models/post.rb]: @mckent05  Thanks for the feedback. Appreciate it.

Comment (shafiuyushawu) [reply app/models/post.rb]: resolved

Comment (Whoistolu) [reply app/models/post.rb]: - Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @shafiuyushawu 

Great job implementing the recommended changes by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by shafiuyushawu** [comment]
@Whoistolu Thanks for the approval

**Pull request merged**